new-frontiers-14/frontier-station-14
# PTK Rework

## About the PR



Largely supercedes #2436, created from suggestions mentioned in comments there.

Gives the PTK a niche purpose: destroying hard rocks.  Walls on asteroids are no longer obstacles to avoid PTK use, but sources of denser ore deposits.  These "supercompacted rock" walls require specialized equipment to destroy (PTK bolts, diamond drills, and/or holopickaxes).  The reflection from walls has been removed.

PTKs in particular now overpenetrate softer rocks, destroying any ore gathered from them.  This requires you to either:
A. be careful with PTK shots, using it to surgically remove higher value ore (that you know about through dwarven intuition or mining analyzers).
B. clear the asteroid in multiple passes, destroying the softer rocks before using the PTK to remove the harder targets
C. use a diamond drill or holopickaxe (generally requiring salvagers to purchase or otherwise source these) and clear the asteroid in one pass

## Why / Balance


Flying around an asteroid with zero threat, blasting rocks away is not good design, and the reflect chance was a bad bandage fix.

## How to test


1. Use the PTK to mine on an asteroid.  Softer rocks should be destroyed leaving nothing, harder rocks (walls) will be destroyed leaving their ore.
2. Use a holopickaxe and diamond drill to mine an asteroid.  All rock types should be destroyed and leave their ores.
3. Use a glaive, pickaxe, any other mining tool you can think of to mine.  Soft rock types should be destroyed, and harder rock types should be unaffected.

## Media


![image](https://github.com/user-attachments/assets/5933cbc3-e0ed-48b8-9a7f-2217adcfd8ce)

## Requirements



## Breaking changes


**Changelog**


:cl:
- tweak: Walls on asteroids can now contain ore denser than regular rocks and require diamond drills, a holopickaxe, or a PTK to mine.
- tweak: PTK bolts destroy ore in softer rock, but "walls" no longer reflect shots.
